File: fontes/backend/java/fenix/src/main/java/com/t2ti/fenix/service/cte/CteCabecalhoService.java
/*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Service relacionado à tabela [CTE_CABECALHO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
package com.t2ti.fenix.service.cte;

import java.util.List;

import javax.persistence.EntityManager;
import javax.persistence.PersistenceContext;
import javax.persistence.Query;

import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import com.t2ti.fenix.model.cte.CteCabecalho;
import com.t2ti.fenix.repository.cte.CteCabecalhoRepository;
import com.t2ti.fenix.model.transiente.Filtro;

@Service
public class CteCabecalhoService {

	@Autowired
	private CteCabecalhoRepository repository;
	
	@PersistenceContext
    private EntityManager entityManager;	
	
	public List<CteCabecalho> consultarLista() {
		return repository.findAll();
	}

	@SuppressWarnings("unchecked")
	public List<CteCabecalho> consultarLista(Filtro filtro) {
		String sql = "select * from CTE_CABECALHO where " + filtro.getWhere();
		Query query = entityManager.createNativeQuery(sql, CteCabecalho.class);
		return query.getResultList();
	}
	
	public CteCabecalho consultarObjeto(Integer id) {
		return repository.findById(id).get();
	}
	
	public CteCabecalho salvar(CteCabecalho objeto) {
		return repository.save(objeto);
	}
		
	public void excluir(Integer id) {
		CteCabecalho objeto = consultarObjeto(id);
		repository.delete(objeto);
	}
	
}

File: fontes/backend/node/fenix-nest/src/cte/cte-cabecalho/cte-cabecalho.service.ts
/*******************************************************************************
Title: T2Ti ERP 3.0                                                                
Description: Service relacionado à tabela [CTE_CABECALHO] 
                                                                                
The MIT License                                                                 
                                                                                
Copyright: Copyright (C) 2021 T2Ti.COM                                          
                                                                                
Permission is hereby granted, free of charge, to any person                     
obtaining a copy of this software and associated documentation                  
files (the "Software"), to deal in the Software without                         
restriction, including without limitation the rights to use,                    
copy, modify, merge, publish, distribute, sublicense, and/or sell               
copies of the Software, and to permit persons to whom the                       
Software is furnished to do so, subject to the following                        
conditions:                                                                     
                                                                                
The above copyright notice and this permission notice shall be                  
included in all copies or substantial portions of the Software.                 
                                                                                
THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,                 
EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES                 
OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND                        
NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT                     
HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,                    
WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING                    
FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR                   
OTHER DEALINGS IN THE SOFTWARE.                                                 
                                                                                
       The author may be contacted at:                                          
           t2ti.com@gmail.com                                                   
                                                                                
@author Albert Eije (alberteije@gmail.com)                    
@version 1.0.0
*******************************************************************************/
import { Injectable } from '@nestjs/common';
import { InjectRepository } from '@nestjs/typeorm';
import { TypeOrmCrudService } from '@nestjsx/crud-typeorm';
import { CteCabecalho } from './cte-cabecalho.entity';
import { DataSource, QueryRunner } from 'typeorm';

@Injectable()
export class CteCabecalhoService extends TypeOrmCrudService<CteCabecalho> {

  constructor(
	private dataSource: DataSource,
    @InjectRepository(CteCabecalho) repository) { super(repository); }

	async persistir(cteCabecalho: CteCabecalho, operacao: string): Promise<CteCabecalho> {
	  let objetoRetorno: CteCabecalho;
	
		const queryRunner = this.dataSource.createQueryRunner();
	
	  await queryRunner.connect();
	  await queryRunner.startTransaction();
	  try {
	    if (operacao == 'A') {
	      await this.excluirFilhos(queryRunner, cteCabecalho.id);
	    }
	    objetoRetorno = await queryRunner.manager.save(cteCabecalho);
	    await queryRunner.commitTransaction();
	  } catch (erro) {
	    await queryRunner.rollbackTransaction();
	    throw (erro);
	  } finally {
	    await queryRunner.release();
	  }
	  return objetoRetorno;
	}
  
	async excluirMestreDetalhe(id: number) {
		const queryRunner = this.dataSource.createQueryRunner();
	
	  await queryRunner.connect();
	  await queryRunner.startTransaction();
	  try {
	    await this.excluirFilhos(queryRunner, id);
	    await queryRunner.query('delete from cteCabecalho where id=' + id);
	    await queryRunner.commitTransaction();
	  } catch (erro) {
	    await queryRunner.rollbackTransaction();
	    throw (erro);
	  } finally {
	    await queryRunner.release();
	  }
	}

	async excluirFilhos(queryRunner: QueryRunner, id: number) {
		await queryRunner.query('delete from CTE_CARGA where ID_CTE_CABECALHO=' + id);
		await queryRunner.query('delete from CTE_DESTINATARIO where ID_CTE_CABECALHO=' + id);
		await queryRunner.query('delete from CTE_LOCAL_ENTREGA where ID_CTE_CABECALHO=' + id);
		await queryRunner.query('delete from CTE_REMETENTE where ID_CTE_CABECALHO=' + id);
		await queryRunner.query('delete from CTE_RODOVIARIO where ID_CTE_CABECALHO=' + id);
		await queryRunner.query('delete from CTE_TOMADOR where ID_CTE_CABECALHO=' + id);
		
	}
	
}